src/ty/ty_check/unify.rs

```diff
@@ -0,0 +1,117 @@
+use crate::ast::Located;
+use crate::ty::{ty_check::Context, Ty, TyError, TyResult};
+
+pub struct Unifier {
```

Comment:
Can you add a `#[derive(Debug)]` to this and to the `Substitution` struct.

Reply:
Sure!